pkg/dns/record.go

```diff
@@ -44,6 +48,35 @@ func newRecord(rule *Rule, flag, domain, remoteIp, ipArea string) (r *Record, er
 	return r, database.DB.Create(r).Error
 }
 
+func FindDomains(c *gin.Context) {
+	var (
+		dnsRecord Record
+		domains   Domains
+		count     int64
+	)
+	res := []string{}
+	if err := c.ShouldBind(&domains); err != nil {
+		c.JSON(400, gin.H{
+			"status": "failed",
+			"error":  err.Error(),
+			"result": nil,
+		})
+	}
+	for i := 0; i < len(domains.Domains); i++ {
+		db := database.DB.Model(&dnsRecord)
+		domain := domains.Domains[i]
+		if err := db.Where("domain like ?", "%"+domain+"%").Count(&count); err != nil {
```

Comment:
```suggestion
		if err := db.Where("domain like ?", "%"+domain+"%").Count(&count).Error; err != nil {
```
以及下面的 append 是不是应该放在 `err == nil`，即数据库查询没有错误的时候执行？`err != nil` 时 `count` 始终为零值 0